File: src/main/java/net/dashflight/data/mfa/MFA_URIDataProvider.java
package net.dashflight.data.mfa;

import java.util.UUID;

public interface MFA_URIDataProvider {

    /**
     * Returns a map of data necessary to build the authenticator URI
     * based on the specified userId. Use this to embed a user's email, username, etc.
     * in the URI.
     */
    BasicUserData getData(UUID userId);

}

File: src/main/java/net/dashflight/data/mfa/MfaUriDataProvider.kt
package net.dashflight.data.mfa

import java.util.*

interface MfaUriDataProvider {
    /**
     * Returns a map of data necessary to build the authenticator URI
     * based on the specified userId. Use this to embed a user's email, username, etc.
     * in the URI.
     */
    fun getData(userId: UUID): BasicUserData
}